### PR TITLE
Moods/cube

### PR DESCRIPTION
**Pull request by maxitect**
# What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Styling
- [ ] Build
- [x] Chore

### Summary

This pull request introduces significant improvements to the 3D cube component, including data refactoring, label management, and interactivity enhancements. It also includes new features and testing to ensure a seamless user experience.

### Updates
- **Cube Component Improvements:**
  - Moved cube quadrants and labels data to external JSON files for improved modularity and maintainability.
  - Fixed label positioning around `Cube.tsx` for a polished appearance.
  - Removed debugging `console.log` statements from `Cube.tsx` for a cleaner codebase.
- **Matrix Toggle Integration:**
  - Added `MatrixToggle.tsx` to manage label changes (e.g., mood vs. priority) dynamically within the cube.
  - Integrated the matrix toggle into the cube to enable switching between mood and priority labels, improving interactivity.
- **Plotly Chart Enhancements:**
  - Added an `onLoaded` callback state to `PlotlyChart.tsx` for tracking when the cube is fully initialized, enhancing responsiveness.
- **Label Dataset Refactoring:**
  - Split the cube label datasets into separate JSON files for moods and priorities, providing clarity and flexibility.
  - Refactored input values and updated IDs and labels for improved accessibility and data handling.
- **Testing:**
  - Added tests for `MatrixToggle.tsx` to verify that it correctly updates the labels within the cube, ensuring functionality meets requirements.
- **Merges:**
  - Incorporated updates from PRs #36, #37, #38, #39, and #40 to align with the latest changes and resolve conflicts.

These updates refine the cube component, improve data organization, and enhance interactivity, creating a more intuitive and maintainable user interface.

## Screenshots

Screen Shot 2024-12-04 at 19 20 10
Screen Shot 2024-12-04 at 19 20 24

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [x] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Delete branch after merge?

- [x] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

tipsywine

### CODE REVIEW

File: src/app/moods/components/data/labels_mood.json
Comment (JasonWarrenUK) [inline src/app/moods/components/data/labels_mood.json:1]: This is pretty

File: src/app/moods/components/MatrixToggle.tsx
Comment (JasonWarrenUK) [inline src/app/moods/components/MatrixToggle.tsx:1]: I don't think this should prevent merging, but it wouldn't be a bad idea to generalise this to a toggle component that can be used elsewhere

**Review by JasonWarrenUK — APPROVED** [review]

File: src/app/moods/components/MatrixToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      </span>
+      <input
+        id="matrix"
+        type="checkbox"
+        checked={isPriorityMatrix}
+        value={isPriorityMatrix ? "priority" : "mood"}
+        onChange={handleMatrixToggle}
+        className="sr-only peer"
+      />
+      <div className="relative w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-twd-primary-purple rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-[2px] after:start-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600"></div>
```
Comment (jackcasstlesjones) [inline src/app/moods/components/MatrixToggle.tsx:35]: please store this in a variable (in this component) for readability, it's too long to be in the returned TSX

File: src/ui/shared/Header.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function Header({ title, description, Icon }: HeaderProps) {
       {/* Right side: Image */}
       <div className="relative w-10 h-10 sm:w-12 sm:h-12">
         <ImageComponent
-          src="/icons/dummy_img.png"
+          src=""
```
Comment (jackcasstlesjones) [inline src/ui/shared/Header.tsx:26]: should there be a prop in Header that is called to populate this?

**Review by jackcasstlesjones — CHANGES_REQUESTED** [review]

File: src/ui/shared/Header.tsx (hunk 2 above)
Comment (maxitect) [reply src/ui/shared/Header.tsx:26]: Agree but right now I'm doubtful that we need a seperate image component to load pngs of icons and why it shouldn't just be another hero icon. I raised this in PR #39, still under consideration I believe

File: src/app/moods/components/MatrixToggle.tsx
Comment (maxitect) [reply src/app/moods/components/MatrixToggle.tsx:1]: Agree, I'll do this this morning

**Review by jackcasstlesjones — APPROVED** [review]

**Pull request merged**